### ISSUE TITLE
Accessibility

### ISSUE THREAD

**Issue by thatblindgeye** [hhhrrrttt222111/developer-portfolio#37]
There are several accessibility features missing and issues that I think would be great to have added in/fixed. Among them are:

- Aria labels need to be added to certain elements, especially ineractive ones. **I already started adding a few to test out the app, so I can take this bullet point on and finish that up.**
- Heading tags need to be fixed up so that levels aren't skipped and are in a more natural order. One example is the EducationCard component using an `<h6` heading followed immediately by `<h4>`, and another example is the SingleBlog component going from an `<h3>` to an `<h6>`
- Semantic HTML needs to be ensured. One example of an issue is the close button on the menu drawer being a `<div>` when it should be a button
- Alt attributes on images need to be looked at. I think for the most part these are good, but there are some cases where I think the attribute should be changed (one example is the image in the Contact section, as that should really have a blank `alt=""` attribute since it seems more decorative than anthing)
- I think certain sections could be tweaked so that the items are in an actual list. Things like the Education section I think could benefit from having each individual education item as part of an actual `<ul>` list.
- Keyboard navigation needs to be improved. Currently it is incredibly difficult if not impossible to tell what element has focus when navigating via keyboard. I think a lot of this can be fixed by just giving the `:focus` pseudoclass the same styling as `:hover`
- Form labels need to be linked to an input. Right now the labels have the `for` attribute, but they aren't actually linked to their respective inputs with an ID.
